internal/dispatch/dispatch.go

```diff
@@ -65,9 +65,21 @@ func CheckDepth(ctx context.Context, req HasMetadata) error {
 	return nil
 }
 
-// CheckRequestToKey converts a check request into a cache key
+// CheckRequestToKey converts a check request into a cache key based on the relation
 func CheckRequestToKey(req *v1.DispatchCheckRequest) string {
-	return fmt.Sprintf("check//%s@%s@%s", tuple.StringONR(req.ObjectAndRelation), tuple.StringONR(req.Subject), req.Metadata.AtRevision)
+	return fmt.Sprintf("check//relation/%s@%s@%s", tuple.StringONR(req.ObjectAndRelation), tuple.StringONR(req.Subject), req.Metadata.AtRevision)
+}
+
+// CheckRequestToKeyWithPossibleCanonical converts a check request into a cache key based possibly
+// on the canonical key, or the relation name if the canonical key is empty
+func CheckRequestToKeyWithPossibleCanonical(req *v1.DispatchCheckRequest, possibleCanonicalKey string) string {
+	if possibleCanonicalKey == "" {
+		return CheckRequestToKey(req)
+	}
+
+	// NOTE: canonical cache keys are only unique *within* a version of a namespace, so they used
+	// essentially as a new relation name in the cache key.
+	return fmt.Sprintf("check//canonical/%s:%s#%s@%s@%s", req.ObjectAndRelation.Namespace, req.ObjectAndRelation.ObjectId, possibleCanonicalKey, tuple.StringONR(req.Subject), req.Metadata.AtRevision)
```

Comment:
Because it would require a significant amount of reworking of the results from those caches (not to mention that lookup is not necessary going to have the full results set since it supports limits)